scalalib/api/src/mill/scalalib/api/ZincWorkerUtil.scala

```diff
@@ -148,6 +146,19 @@ trait ZincWorkerUtil {
       all.filter(v => v.nonEmpty && v >= versionParts.take(v.length)).map(_.mkString(".") + "-")
     (plus ++ minus).distinct.toSeq
   }
+
+  /**
+   * Helper to create the Mill plugin artifact structure that is expected by Mill.
+   *
+   * Ex. If your plugin name is example and you're using 0.10.0 this would create:
+   *   - example_mill0.10
+   *
+   * @param name the name of your plugin
+   * @param millVersion the Mill version you're targeting
+   * @return
+   */
+  def pluginArtifact(name: String, millVersion: String) =
+    s"${name}_mill${scalaBinaryVersion(millVersion)}"
```

Comment:
I think the same makes it hard to guess, which part of the artifact(name/id) is the result of this `def`. What about just having a `def` that only calculated the platform suffix?

`scalaBinaryVersion` is specific for Scala versions, e.g. versions starting with `3` are handled differently than those starting with `2`. Here we should keep logic specific to Mill plugins, e.g. when starting with `0`, we use the major+minor. But for major `1`, we probably want to switch to a more stable platform which only includes the major.

Reply:
> I think the name makes it hard to guess, which part of the artifact(name/id) is the result of this def. What about just having a def that only calculated the platform suffix?

I'm not fully sure I understand what you mean? So you're saying that the `def` here should only calculate `_mill0.10` part? So then the user would just use this with their name locally to put together the name? If so, I still think it's nicer to have a method that does it all, so then the user doesn't have to think about it at all, just pass in their name and the mill version, and boom, done.

> scalaBinaryVersion is specific for Scala versions, e.g. versions starting with 3 are handled differently than those starting with 2. Here we should keep logic specific to Mill plugins, e.g. when starting with 0, we use the major+minor. But for major 1, we probably want to switch to a more stable platform which only includes the major.

I can change this back and instead create a `millBinaryVersion` maybe? And this is only used for this purpose? wdyt?

Reply:
> > I think the name makes it hard to guess, which part of the artifact(name/id) is the result of this def. What about just having a def that only calculated the platform suffix?
> 
> I'm not fully sure I understand what you mean? So you're saying that the `def` here should only calculate `_mill0.10` part? So then the user would just use this with their name locally to put together the name? If so, I still think it's nicer to have a method that does it all, so then the user doesn't have to think about it at all, just pass in their name and the mill version, and boom, done.

Sorry for being not clear enough. I think a `def pluginArtifact` leaves too much doubt what it is returning to the user. The pure name ( `foo`), or the full artifactId (`foo_mill0.10_2.13.8`) or some partial full name (`foo_mill_0.10`). Maybe the name is too vague? I wonder where this method is to be used?

Shouldn't we instead just set the `JavaModule#platformSuffix`? And always render it as part of the `artifactSuffix`, which is currently done identically in `ScalaJSModule` and `ScalaNativeModule` ?

We could even went ahead and introduce a `MillPluginModule`, that also contains the logic of inferring the correct platform suffix based on the Mill version (or some enum-like selection).


> > scalaBinaryVersion is specific for Scala versions, e.g. versions starting with 3 are handled differently than those starting with 2. Here we should keep logic specific to Mill plugins, e.g. when starting with 0, we use the major+minor. But for major 1, we probably want to switch to a more stable platform which only includes the major.
> 
> I can change this back and instead create a `millBinaryVersion` maybe? And this is only used for this purpose? wdyt?

Yeah, a `def millBinaryVersion` is what I had in mind.

Reply:
Ahhhh yes, good idea. I like the idea of a `MillPluginModule`. I'll close this and pursue that. I didn't even think about the `JavaModule#platformSuffix` which also make sense.